File: pinkmoe_index/src/hooks/shop/shop.ts
/*
 * @Author: coderzhaolu && izhaicy@163.com
 * @Date: 2022-07-24 08:40:11
 * @LastEditors: coderzhaolu && izhaicy@163.com
 * @LastEditTime: 2022-08-11 07:21:23
 * @FilePath: /pinkmoe_index/src/hooks/shop/shop.ts
 * @Description: https://github.com/Coder-ZhaoLu/pinkmoe   (如需用于商业用途或者二开，请联系作者捐助任意金额即可)
 * QQ:2419857357;支付宝:13135986153
 * Copyright (c) 2022 by coderzhaolu, All Rights Reserved.
 */
import { ResPage } from '/@/api/common/types';
import { ResPost } from '/@/api/home/types';
import { ReqBbsPost } from '/@/api/post/types';
import { useUserStore } from '/@/store';

export const useShop = () => {
  const postList = ref<ResPage<Array<ResPost>>>();
  const { proxy } = getCurrentInstance();
  const shopMenu = ref();
  const sort = ref<any>([
    {
      title: '按最新',
      type: 'updated_at',
    },
    {
      title: '按标题',
      type: 'title',
    },
    {
      title: '按作者',
      type: 'author',
    },
    {
      title: '按查看',
      type: 'view',
    },
  ]);
  const auth = useUserStore();
  const formParams: ReqBbsPost = reactive({
    userId: auth?.userInfo?.uuid,
    category: '',
    type: '',
    page: 1,
    pageSize: 12,
    orderKey: 'updated_at',
    desc: true,
  });

  const loading = ref<boolean>(false);
  const hasMore = ref<boolean>(true);

  const scrollMenu = (isRight) => {
    shopMenu.value.scrollTo({
      // y方向坐标800px（代码中不需要写明单位）
      left: isRight ? shopMenu.value.scrollLeft + 140 : shopMenu.value.scrollLeft - 140,
      // 滚动方式是平滑滚动 默认是auto 即instant 直接跳到目标位置  不常用
      behavior: 'smooth',
    });
  };

  return {
    postList,
    sort,
    formParams,
    hasMore,
    loading,
    auth,
    scrollMenu,
    shopMenu,
  };
};

File: pinkmoe_server/model/response/shop.go
/*
 * @Author: coderzhaolu && izhaicy@163.com
 * @Date: 2022-06-22 11:13:07
 * @LastEditors: coderzhaolu && izhaicy@163.com
 * @LastEditTime: 2022-08-07 08:54:23
 * @FilePath: /pinkmoe_server/model/response/home.go
 * @Description: https://github.com/Coder-ZhaoLu/pinkmoe 
 * 问题反馈qq群:749150798
 * xanaduCms程序上所有内容(包括但不限于 文字，图片，代码等)均为指针科技原创所有，采用请注意许可
 * 请遵循 “非商业用途” 协议。商业网站或未授权媒体不得复制内容，如需用于商业用途或者二开，请联系作者捐助任意金额即可，我们将保存所有权利。
 * Copyright (c) 2022 by 指针科技, All Rights Reserved.
 */
package response

import "server/model"

type ShopConfig struct {
	Popular   []string  `json:"popular"`
	Recommend []string  `json:"recommend"`
	Cms       []ShopCms `json:"cms"`
}

type ShopCms struct {
	Category string `json:"category"`
	Style    int    `json:"style"`
}

type ShopSettings struct {
	ParsingUrl string `json:"parsingUrl"`
}

type ShopContent struct {
	Slug  string          `json:"slug"`  // 菜单标识
	Name  string          `json:"name"`  // 菜单名称
	Icon  string          `json:"icon"`  // 菜单图标
	Sort  int             `json:"sort"`  // 排序标记
	Style int             `json:"style"` // 分类模块显示类型
	Goods []model.XdGoods `json:"goods"` // 文章
}

type Shop struct {
	Popular      []model.XdGoods         `json:"popular"`
	ShopCategory []model.XdGoodsCategory `json:"shopCategory"`
	Content      []ShopContent           `json:"content"`
}
